src/wireviz.py

```diff
@@ -327,15 +335,17 @@ def __post_init__(self):
         self.ports_right = False
         self.loops = []
 
-        if self.pinout:
-            if self.pincount is not None:
-                raise Exception('You cannot specify both pinout and pincount')
-            else:
+        if self.pincount is None:
+            if self.pinout:
                 self.pincount = len(self.pinout)
-        else:
-            if not self.pincount:
-                self.pincount = 1
-            self.pinout = ['',] * self.pincount
+            elif self.pinnumbers:
+                self.pincount = len(self.pinnumbers)
+            else:
+                raise Exception('You need to specify at least one, pincount, pinout or pincount')
```

Comment:
I guess one of these should be pinnumbers

Reply:
sure. fixed.